polaris-react/src/components/AppProvider/AppProvider.tsx

```diff
@@ -24,6 +24,39 @@ import type {FeaturesConfig} from '../../utilities/features';
 import './AppProvider.scss';
 import './global.scss';
 
+function measureScrollbars() {
+  const maxScrollbarWidth = 20;
+  const overflowParentHeight = 30;
+  const overflowChildHeight = 40;
+  const parentEl = document.createElement('div');
+  parentEl.setAttribute(
+    'style',
+    `position: absolute; opacity: 0; transform: translate3d(-9999px, -9999px, 0); pointer-events: none; width:${overflowParentHeight}px; height:${overflowParentHeight}px;`,
+  );
+
+  const child = document.createElement('div');
+  child.setAttribute(
+    'style',
+    `width:100%; height: ${overflowChildHeight}; overflow:scroll`,
+  );
+  parentEl.appendChild(child);
+  document.body.appendChild(parentEl);
+
+  let scrollbarWidth =
+    overflowParentHeight - (parentEl.firstChild as HTMLDivElement).clientWidth;
+
+  if (scrollbarWidth > maxScrollbarWidth) {
+    scrollbarWidth = maxScrollbarWidth;
+  }
```

Comment:
Why are we wanting to enforce a maximum scrollbar width?

I think an alternative to solve this problem, so you don't need to use `let` and could possibly simplify this is be changing the `scrollbarWidth` variable name to `calculatedScrollbarWidth` or something like that, then on line 54, where this const is being used do `Math.min(calculatedScrollbarWidth, MAX_SCROLLBAR_WIDTH)`.

Reply:
> Why are we wanting to enforce a maximum scrollbar width?

This is purely paranoia on my end 😅 What if something with the calculation goes wrong on some platform and the scrollbar width becomes `infinity` or some shenanigans like that. This would prevent that.